OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -103,7 +105,7 @@ public void Damaged(Actor self, AttackInfo e)
 			}
 
 			// not a lot we can do about things we can't hurt... although maybe we should automatically run away?
-			if (!attack.HasAnyValidWeapons(Target.FromActor(attacker)))
+			if (!attackBases.Any(a => !a.IsTraitDisabled && a.HasAnyValidWeapons(Target.FromActor(attacker))))
```

Comment:
Minor style nit: these might be easier to read as `if (attackBases.All(a => a.IsTraitDisabled || !a.HasAnyValidWeapons(Target.FromActor(attacker))))` (removing the double negatives).